temporal-test-server/src/main/java/io/temporal/internal/testservice/SelfAdvancingTimerImpl.java

```diff
@@ -207,27 +206,39 @@ public TimerLockHandle(LockEvent event) {
 
     @Override
     public void unlock() {
-      unlock(null);
+      lock.lock();
+      try {
+        unlockFromHandleLocked(event.caller + "[⎌]"); // caller releases its own lock
+        condition.signal();
+      } finally {
+        lock.unlock();
+      }
     }
 
     @Override
     public void unlock(String caller) {
       lock.lock();
       try {
-        unlockFromHandleLocked();
+        unlockFromHandleLocked(
+            "{-"
+                + caller
+                + "}"
+                + " {+"
+                + event.caller
+                + "}"); // the caller releases lock taken by another caller using the same handle
         condition.signal();
       } finally {
         lock.unlock();
       }
```

Comment:
It looks like both of these `unlock` overloads could share one impl just passing the different string in.

A comment on it would be nice too about why it locks/unlocks in the same method